rc2014 - Added the correct company name for the RC2014 Kits.

feilipu opened the pull request:
[RFC2795 LTD](https://find-and-update.company-information.service.gov.uk/company/10159326/) - Company number 10159326

Spencer Owen owns the RC2014 trademark, and trades as RFC2795 LTD.

cuavas commented:
UK company registrations are case-folded to uppercase, but the company usually seems to render it as “RFC2795 Ltd” as seen on this page: https://z80kits.com/about/ (“z80kits.com is a trading name of RFC2795 Ltd”)

feilipu commented:
Then that would be even more correct. 👍

feilipu commented:
Updated the company name.

Also updated the introduction year based on date (October 2016) identified in the [schematic](http://rc2014.co.uk/wp-content/uploads/2016/10/RC2014-Mini.pdf) and [Assembly Guide](http://rc2014.co.uk/wp-content/uploads/2016/07/RC2014-Mini-Assembly-Guide.pdf).
